### ISSUE TITLE
Compatibility Challenges with Zig Language Extras on Windows 10 LTSC Using MinGW-W64 and Scoop and Ubuntu 23.10

### ISSUE THREAD

**Issue by punowo** [ianic/zig-language-extras#5]
Hello. First of all thank you for this extension. I've wanted something like this for a while. https://github.com/ziglang/vscode-zig/issues/158

I'm having some trouble getting this to work and this might be due to my somewhat unorthodox setup, I say this because from what I've seen most people just use msvc on Windows. 

I am on Windows 10 LTSC and I pull `mingw-winlibs-llvm-ucrt-mcf` ( `gcc`, `gdb` ), `zig` and `zls` from scoop ( https://scoop.sh/ ). With C/C++ I just use handle everything through the cmake vscode plugin which has the ability to initialize projects, debug cmake files and even `Cmake: Debug` which just works no matter what OS I'm on as long as there a C/C++ compiler present. 
```
PS C:\Users\pun> gcc --version
gcc.exe (MinGW-W64 x86_64-ucrt-mcf-seh, built by Brecht Sanders) 13.2.0
Copyright (C) 2023 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

PS C:\Users\pun> gdb --version
GNU gdb (GDB for MinGW-W64 x86_64, built by Brecht Sanders) 13.2
Copyright (C) 2023 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

PS C:\Users\pun> zig version
0.11.0
```
I've made a simple `hello` type test project. 

![folder_structure](https://github.com/ianic/zig-language-extras/assets/96526435/ac86ae2b-0a53-41bb-8ad2-9ef3f496be37)

Here are the results for each respective `"zig-language-extras.debugType"` option: 

`cppvsdbg`

![cppvsdbg](https://github.com/ianic/zig-language-extras/assets/96526435/fd5acd16-a07c-4f62-ae16-69e87ab6e69f)

`gdb`

![gdb](https://github.com/ianic/zig-language-extras/assets/96526435/ee5b5288-d4c0-403c-ac85-efd49429e593)

`lldb`

![lldb](https://github.com/ianic/zig-language-extras/assets/96526435/ce38c339-dc0e-40cf-ae15-8bb0c3622fe6)

I can getting working by having the project name be `main` with the default option for `win32`, `cppvsdbg`. 

![works](https://github.com/ianic/zig-language-extras/assets/96526435/240fac4e-3b46-4baa-ad9f-b77b5a696f70)

**Comment by ianic** [comment]
Uf, Windows... never tried it there.

Please look at #3 where @Niakr1s introduced Windows support.
Maybe you can set 'zig-language-extras.debugType' to your debugger.

**Comment by ianic** [comment]
Can you manually build launch config, this is what [extension does](https://github.com/ianic/zig-language-extras/blob/6fd60485e3e7698eab01f8e87b96b81a2dc04107/src/extension.ts#L104).
Then start with launch config from extensions and then tweak it to work with different debuggers. 

binPath can be found in extension [output](https://github.com/ianic/zig-language-extras/blob/6fd60485e3e7698eab01f8e87b96b81a2dc04107/src/extension.ts#L98).

**Comment by ianic** [comment]
Thanks for your effort in bringing this up.

When debugging binary, extension tries to decide on binary name in zig-out/bin folder. That should be name of the file or for main.zig name of the parent folder (skipping src). I found a [problem](https://github.com/ianic/zig-language-extras/commit/b75fd1be47bc4682af861702bb2ffe779feddfe4) in that logic. 

Could you please try with the latest 0.62 version.

**Comment by ghost** [comment]
I don't have access to my windows machine but I'm pretty sure you can close this. The problem is fixed for me on my Ubuntu 23.10 machine. Thanks for the update.
